src/libexpr/eval.cc

```diff
@@ -2525,12 +2525,17 @@ Strings EvalSettings::getDefaultNixPath()
 {
     Strings res;
     auto add = [&](const Path & p, const std::string & s = std::string()) {
-        if (pathExists(p)) {
-            if (s.empty()) {
-                res.push_back(p);
-            } else {
-                res.push_back(s + "=" + p);
+        try {
+            if (pathExists(p)) {
+                if (s.empty()) {
+                    res.push_back(p);
+                } else {
+                    res.push_back(s + "=" + p);
+                }
             }
+        } catch (SysError & e) {
+            // swallow EPERM
+            if (e.errNo != EPERM) throw;
```

Comment:
Would it make sense to have a variant of `pathExists` that treats `EPERM` as `ENOENT` (either another function or a flag argument to it)? This seems to be a recurring thing with the Darwin sandbox, and we often don't care about the distinction so it feels like it could make sense